src/uct/ib/base/ib_device.c

```diff
@@ -176,6 +195,79 @@ static void uct_ib_device_get_locality(const char *dev_name,
     *numa_node = (status == UCS_OK) ? n : -1;
 }
 
+static void
+uct_ib_device_async_event_dispatch(uct_ib_device_t *dev,
+                                   const uct_ib_async_event_t *event)
+{
+    uct_ib_async_event_val_t *entry;
+    khiter_t iter;
+    int ret;
+
+    ucs_spin_lock(&dev->async_event_lock);
+    iter  = kh_put(uct_ib_async_event, &dev->async_events_hash, *event, &ret);
+    entry = &kh_value(&dev->async_events_hash, iter);
```

Comment:
minor: this entry initialization can be moved to `if (ret == UCS_KH_PUT_KEY_PRESENT)`